src/fides/api/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -1918,3 +1938,188 @@ def _process_privacy_request_restart(
+        request_task.id,
+    )
+
+    # Requeuing the particular task now that the callback is received
+    log_task_queued(request_task)
+    task_function(request_task).delay(privacy_request.id, request_task.id)
+
+    return {"task_queued": True}
+
+
+def task_function(request_task: RequestTask) -> Callable:
+    mapping = {
+        ActionType.access: run_access_node,
+        ActionType.erasure: run_erasure_node,
+        ActionType.consent: run_consent_node,
+    }
+    return mapping[request_task.action_type]
+
+
+@router.post(
+    PRIVACY_REQUEST_TASK_CALLBACK,
+    dependencies=[
+        Security(verify_oauth_client, scopes=[PRIVACY_REQUEST_CALLBACK_RESUME])
+    ],
+    response_model=Dict,
+)
+def task_callback(
+    privacy_request_id: str,
+    task_id: str,
+    *,
+    db: Session = Depends(deps.get_db),
+) -> Dict:
+    """
+    Early callback POC endpoint - just for experimenting!
+
+    Here I'm just demonstrating we can run an individual task from this point provided its upstream tasks are complete and the task
+    itself is in the appropriate state.
+
+    Individual task functions don't yet support working with any input data other than data dependencies described in the dataset
+    """
+    try:
+        privacy_request, request_task, _ = run_prerequisite_task_checks(
+            db, privacy_request_id, task_id
+        )
+    except (PrivacyRequestNotFound, RequestTaskNotFound) as exc:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail=str(exc),
+        )
+
+    if privacy_request.status != PrivacyRequestStatus.in_processing:
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail=f"Callback failed. Cannot run {request_task.action_type.value} task {request_task.id} with privacy request of status {request_task.status.value}",
+        )
+
+    if request_task.status != ExecutionLogStatus.paused:
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail=f"Callback failed. Cannot run {request_task.action_type.value} task {request_task.id} with status {request_task.status.value}",
+        )
+
+    logger.info(
+        "Callback received for {} task {} {}",
+        request_task.action_type.value,
+        request_task.collection_address,
+        request_task.id,
+    )
+
+    # Mark that the callback was received on the task itself - just experimenting here. We could also save data
+    # to the request task itself in another column
+    request_task.callback_succeeded = True
+    request_task.update_status(db, ExecutionLogStatus.pending)
+    request_task.save(db)
+
+    if request_task.action_type == ActionType.access:
+        # Requeuing the particular task now that the callback is received
+        log_task_queued(request_task)
+        run_access_node.delay(privacy_request.id, request_task.id)
+
+    # ... handle other cases
+
+    return {"task_queued": True}
```

Comment:
Experimental endpoint for async connectors - removed in https://github.com/ethyca/fides/pull/4760/files to keep that work focused on DSR 3.0